src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,144 @@
+        user.setPassword("validPass");
+        assertThrows(InvalidUserDataException.class, () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    void register_negativeAge_notOk() {
+        user.setAge(-5);
+        user.setLogin("validLogin");
+        user.setPassword("validePasssword");
+        assertThrows(InvalidUserDataException.class, () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    void register_ageEquals18_ok() {
+        user.setAge(18);
+        user.setLogin("valLog");
+        user.setPassword("valPas");
+        User registeredUser = registrationServiceImpl.register(user);
+        assertEquals(user, registeredUser);
+    }
+
+    @Test
+    void register_LoginExist_notOk() {
+        User existingUser = new User();
+        existingUser.setLogin("validLogin");
+        Storage.people.add(existingUser);
+        user.setAge(20);
+        user.setLogin("validLogin");
+        user.setPassword("validPass");
+        assertThrows(InvalidUserDataException.class, () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    void registre_LoginLength_notOk() {
+        user.setAge(20);
+        user.setLogin("log");
+        user.setPassword("validPass");
+        assertThrows(InvalidUserDataException.class, () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    void register_LoginLengthEdgeCase_notOk() {
+        user.setAge(18);
+        user.setLogin("valid");
+        user.setPassword("validP");
+        assertThrows(InvalidUserDataException.class, () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    void register_LoginLengthEquals_ok() {
+        user.setAge(18);
+        user.setLogin("validL");
+        user.setPassword("validP");
+        User registeredUser = registrationServiceImpl.register(user);
+        assertEquals(user, registeredUser);
+    }
+
+    @Test
+    void register_PasswordLength_notOk() {
+        user.setAge(20);
+        user.setLogin("validLogin");
+        user.setPassword("passp");
+        assertThrows(InvalidUserDataException.class, () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    void register_PasswordLength_Ok() {
+        user.setAge(18);
+        user.setLogin("validLogin");
+        user.setPassword("validP");
+        User registeredUser = registrationServiceImpl.register(user);
+        assertEquals(user, registeredUser);
+    }
+
+    @Test
+    void register_PasswordLengthEdgeCase_notOk() {
+        user.setAge(20);
+        user.setLogin("validLogin");
+        user.setPassword("valid");
+        assertThrows(InvalidUserDataException.class, () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    void registr_PasswordLengthEqalse_Ok() {
+        user.setAge(20);
+        user.setLogin("validLogin");
+        user.setPassword("validP");
+        User registeredUser = registrationServiceImpl.register(user);
+        assertEquals(user, registeredUser);
```

Comment:
The test method 'registr_PasswordLengthEqalse_Ok' contains a typo ('Eqalse' instead of 'Equals') and does not clearly state what is being tested. Consider renaming it to 'register_passwordMinLength_ok' if it tests the minimum valid password length.